hazelcast/test/src/map/ClientMapTest.cpp

```diff
@@ -3370,6 +3370,31 @@ namespace hazelcast {
 
                 ClientMapTest<TypeParam>::imap->removeInterceptor(interceptorId);
             }
+
+            TYPED_TEST(ClientMapTest, testJsonPutGet) {
+                IMap<string, serialization::json::HazelcastJsonValue> map = ClientMapTest<TypeParam>::client->template getMap<std::string, serialization::json::HazelcastJsonValue>(
+                        ClientMapTest<TypeParam>::getTestName());
+                serialization::json::HazelcastJsonValue value("{ \"age\": 4 }");
+                map.put("item1", value);
+                boost::shared_ptr<serialization::json::HazelcastJsonValue> retrieved = map.get("item1");
+
+                ASSERT_EQ_PTR(value, retrieved.get(), serialization::json::HazelcastJsonValue);
+            }
+
+            TYPED_TEST(ClientMapTest, testQueryOverJsonObject) {
+                IMap<string, serialization::json::HazelcastJsonValue> map = ClientMapTest<TypeParam>::client->template getMap<std::string, serialization::json::HazelcastJsonValue>(
+                        ClientMapTest<TypeParam>::getTestName());
+                serialization::json::HazelcastJsonValue young("{ \"age\": 4 }");
+                serialization::json::HazelcastJsonValue old("{ \"age\": 20 }");
+                map.put("item1", young);
+                map.put("item1", old);
+
+                // Get the objects whose age is less than 6
+                std::vector<serialization::json::HazelcastJsonValue> result = map.values(
+                        query::GreaterLessPredicate<int>("a", 6, false, true));
```

Comment:
Does this test pass ? Json does not have a "a" field . If this test is passing we need to check why ?

Reply:
I think there is no bug but i mistakenly inserted the second value with the same key as item1 and it overwrote the other value. And it was due to this mistake the test passed :) Corrected it now. Also added a check to verify that map has 2 entries.